packages/mermaid/src/diagrams/git/gitGraphAst.ts

```diff
@@ -11,17 +11,43 @@ import {
   setDiagramTitle,
   getDiagramTitle,
 } from '../common/commonDb.js';
+import defaultConfig from '../../defaultConfig.js';
+import { CommitType } from './gitGraphTypes.js';
 
-let mainBranchName = getConfig().gitGraph.mainBranchName;
-let mainBranchOrder = getConfig().gitGraph.mainBranchOrder;
-let commits = {};
-let head = null;
-let branchesConfig = {};
+// TODO
+type CommitParentGraph = {
+  // First element is null, rest is parent
+};
+type BranchConfig = {
+  name: string;
+  order: number;
+};
+type DiagramOrientation = 'LR' | 'TB';
+type Commit = {
+  id: string;
```

Comment:
```suggestion
interface Commit {
  id: string;
```

Please use interface, not types.